Tests/AdyenTests/Card Tests/CardComponentTests.swift

```diff
@@ -1024,6 +1024,64 @@ class CardComponentTests: XCTestCase {
         waitForExpectations(timeout: 20, handler: nil)
     }
     
+    func testBrazilSSNs() {
+        let method = CardPaymentMethod(type: "bcmc", name: "Test name", fundingSource: .credit, brands: ["visa", "amex", "mc", "elo"])
+        let config = CardComponent.Configuration(socialSecurityNumberMode: .auto)
+        let cardTypeProviderMock = BinInfoProviderMock()
+        cardTypeProviderMock.onFetch = {
+            $0(BinLookupResponse(brands: [CardBrand(type: .elo)],
+                                 issuingCountryCode: "BR",
+                                 showSocialSecurityNumber: true))
+        }
+
+        let sut = CardComponent(paymentMethod: method,
+                                apiContext: Dummy.context,
+                                configuration: config,
+                                style: .init(),
+                                cardPublicKeyProvider: CardPublicKeyProviderMock(),
+                                binProvider: cardTypeProviderMock)
+        UIApplication.shared.keyWindow?.rootViewController = sut.viewController
+
+        let delegate = PaymentComponentDelegateMock()
+        sut.delegate = delegate
+
+        let delegateExpectation = expectation(description: "PaymentComponentDelegate must be called when submit button is clicked.")
+        delegate.onDidFail = { error, component in XCTFail("should not fail") }
+        delegate.onDidSubmit = { data, component in
+            XCTAssertTrue(component === sut)
+            let paymentDetails = data.paymentMethod as? CardDetails
+            XCTAssertNotNil(paymentDetails)
+            XCTAssertEqual(paymentDetails?.socialSecurityNumber, "12312312312")
+
+            sut.stopLoadingIfNeeded()
+            delegateExpectation.fulfill()
+        }
+
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + .seconds(1)) {
```

Comment:
you can do `wait(for: .seconds(1))` and continue code linearly :-)

Reply:
think this is a cleaner and more elegant way as used on all other methods instead of plain wait call :)

Reply:
`wait()` function was added to get ride of the ugly multiple nested closures, tbh I have no idea what you mean by "cleaner and more elegant way" :-D